includes/namespace.php

```diff
@@ -219,3 +220,77 @@ function get_plugin_instance() {
 	global $web_stories;
 	return $web_stories;
 }
+
+/**
+ * Append result of internal request to REST API for purpose of preloading data to be attached to a page.
+ * Expected to be called in the context of `array_reduce`.
+ *
+ * Like rest_preload_api_request() in core, but removes trailing slashes.
+ *
+ * @see \rest_preload_api_request
+ * @link https://core.trac.wordpress.org/ticket/51636
+ *
+ * @since 1.1.0
+ *
+ * @param array        $memo Reduce accumulator.
+ * @param string|array $path REST API path to preload.
+ * @return array Modified reduce accumulator.
+ */
+function rest_preload_api_request( $memo, $path ) {
+	// array_reduce() doesn't support passing an array in PHP 5.2,
+	// so we need to make sure we start with one.
+	if ( ! is_array( $memo ) ) {
+		$memo = [];
+	}
+
+	if ( empty( $path ) ) {
+		return $memo;
+	}
+
+	$method = 'GET';
+	if ( is_array( $path ) && 2 === count( $path ) ) {
+		$method = end( $path );
+		$path   = reset( $path );
+
+		if ( ! in_array( $method, [ 'GET', 'OPTIONS' ], true ) ) {
+			$method = 'GET';
+		}
+	}
+
+	$path_parts = wp_parse_url( $path );
+	if ( false === $path_parts ) {
+		return $memo;
+	}
+
+	$request = new WP_REST_Request( $method, untrailingslashit( $path_parts['path'] ) );
```

Comment:
Is this the only line difference? 

After this line, we could just call
```
return \rest_preload_api_request( $memo, $path );
```

Reply:
Yes. Good idea to call the original function 👍

Reply:
Or well, I'd still need to somehow merge the resulting arrays 🤔